microsoft/testsuites/vm_extensions/run_command.py

```diff
@@ -306,3 +309,49 @@ def verify_sas_uri_script_run(
         _create_and_verify_extension_run(
             node, settings, f"ls '{test_file}'", 0, message
         )
+
+    @TestCaseMetadata(
+        description="""
+        Runs the Run Command v2 VM extension with a timeout of 0.1 seconds.
+        """,
+        priority=3,
+        requirement=simple_requirement(supported_features=[AzureExtension]),
+    )
+    def verify_script_run_with_timeout(self, log: Logger, node: Node) -> None:
+        test_file = "/tmp/rcv2timeout.txt"
+        settings = {
+            "source": {
+                "CommandId": "RunShellScript",
+                "script": f"sleep 0.1; touch {test_file}",
+            },
+            "timeoutInSeconds": 1,
+        }
+        message = f"File {test_file} was not created on the test machine"
+
+        _create_and_verify_extension_run(
+            node, settings, f"ls '{test_file}'", 0, message
+        )
+
+    @TestCaseMetadata(
+        description="""
+        Runs the Run Command v2 VM extension with a timeout of 1 second.
+        """,
+        priority=3,
+        requirement=simple_requirement(supported_features=[AzureExtension]),
+    )
+    def verify_script_run_with_timeout_failed(self, log: Logger, node: Node) -> None:
+        test_file = "/tmp/rcv2timeout.txt"
+        settings = {
+            "source": {
+                "CommandId": "RunShellScript",
+                "script": f"sleep 1; touch {test_file}",
```

Comment:
Make it greater than timeout value, so it's more reliable to be failed. At least 1.5 seconds?